File: DesktopSample/API/Section/Charting/VB.NET/Charting/ViewerForm.vb
Imports System.Data.SQLite
Imports System.Text
Imports System.Xml
Imports GrapeCity.ActiveReports
Imports GrapeCity.ActiveReports.SectionReportModel

Public Class ViewerForm
	Inherits System.Windows.Forms.Form

	Public Sub New()
		MyBase.New()
		Encoding.RegisterProvider(CodePagesEncodingProvider.Instance)
		'必要なデザイナ変数です。
		InitializeComponent()

		' 
		'TODO：InitializeComponentの呼び出しの後に、それはコンストラクタです。コードを追加してください。
	End Sub

	' Windows フォーム デザイナ サポートに必要です。
	Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
		If disposing Then
			If Not (components Is Nothing) Then
				components.Dispose()
			End If
		End If
		MyBase.Dispose(disposing)
	End Sub

	Private Sub ViewerForm_Load(ByVal sender As System.Object, ByVal e As EventArgs) Handles MyBase.Load
		cboStyle.Items.Add(My.Resources.TwoDBarChart)
		cboStyle.Items.Add(My.Resources.ThreeDPieChart)
		cboStyle.Items.Add(My.Resources.ThreeDBarChart)
		cboStyle.Items.Add(My.Resources.FinanceChart)
		cboStyle.Items.Add(My.Resources.StackedAreaChart)

		' 「グラフの種類」コンボボックスの初期選択状態を設定します。
		cboStyle.SelectedIndex = 0
	End Sub

	Private Sub cboList_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As EventArgs) Handles cboStyle.SelectedIndexChanged
		' 「折れ線グラフ」選択時のみ、カスタムプロパティコンボボックスを有効にします。
		Call SetCustomProperties(Me.cboStyle.SelectedIndex)
	End Sub

	Private Sub btnReport_Click(ByVal sender As System.Object, ByVal e As EventArgs) Handles btnReport.Click

		Dim rpt As New SectionReport()
		Try
			' 「グラフの種類」コンボボックスにあわせて、プレビュー表示します。
			Select Case cboStyle.SelectedIndex
				Case 0  ' 2D棒グラフ
					rpt.LoadLayout(XmlReader.Create(My.Resources.rpt2DBar))
				Case 1  ' 3D円グラフ
					rpt.LoadLayout(XmlReader.Create(My.Resources.rpt3DPie))
					If cboCustom.SelectedIndex = 0 Then
						DirectCast(rpt.Sections("Detail").Controls("ChartSalesCategories"), ChartControl).Series(0).Properties("Clockwise") = True
					Else
						DirectCast(rpt.Sections("Detail").Controls("ChartSalesCategories"), ChartControl).Series(0).Properties("Clockwise") = False
					End If
				Case 2  '  3D棒グラフ
					rpt.AddAssembly(GetType(SQLiteConnection).Assembly)
					rpt.LoadLayout(XmlReader.Create(My.Resources.rpt3DBar))
				Case 3  '  ファイナンスチャート
					rpt.LoadLayout(XmlReader.Create(My.Resources.rptCandle))
				Case 4  '  積層エリアグラフ
					rpt.LoadLayout(XmlReader.Create(My.Resources.rptStackedArea))
			End Select
			arvMain.LoadDocument(DirectCast(rpt, SectionReport))
		Catch ex As Exception
			MessageBox.Show(ex.ToString)
		End Try

	End Sub

	Private Sub SetCustomProperties(ByVal iGraphStype As Integer)
		' 選択候補をクリアします。
		cboCustom.Items.Clear()

		' 選択候補を追加し、選択状態にします。
		Select Case iGraphStype
			Case 1    '  2D円グラフ
				' 
				'可視状態を変更します。
				cboCustom.Visible = True
				lblCustom.Visible = True

				cboCustom.Items.Add(My.Resources.Clockwise)
				cboCustom.Items.Add(My.Resources.Counterclockwise)

				cboCustom.SelectedIndex = 1

				'
				'ラベルを設定します。
				lblCustom.Text = My.Resources.DirectionOfRotation
			Case Else   '  その他
				'不可視状態にします。
				cboCustom.Visible = False
				lblCustom.Visible = False
		End Select

		Application.DoEvents()
	End Sub

End Class

File: DesktopSample/API/Section/Charting/C#/Charting/ViewerForm.cs
using GrapeCity.ActiveReports;
using GrapeCity.ActiveReports.SectionReportModel;
using System;
using System.Data.SQLite;
using System.Text;
using System.Windows.Forms;
using System.Xml;

namespace ActiveReports.Samples.Charting
{
	public partial class ViewerForm : System.Windows.Forms.Form
	{
		public ViewerForm()
		{
			//
			//Windows フォーム デザイナ サポートに必要です。
			//
			InitializeComponent();

			// TODO：InitializeComponentの呼び出しの後に、それはコンストラクタです。コードを追加してください
			//
		}
		
		[STAThread]
		static void Main()
		{
			Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
			Application.EnableVisualStyles();
			Application.SetCompatibleTextRenderingDefault(false);
#if NET6_0_OR_GREATER
			Application.SetHighDpiMode(HighDpiMode.DpiUnawareGdiScaled);
#endif
			Application.Run(new ViewerForm());
		}

		/// <summary>
		/// ViewerForm上のViewerコントロールにレポートをロードします。
		/// ViewerコントロールがViewerFormインスタンス上にロードされたとき、Loadイベントは発生します。
		/// </summary>
		private void ViewerForm_Load(object sender, EventArgs e)
		{
			cboStyle.Items.Add(Properties.Resources.TwoDBarChart);
			cboStyle.Items.Add(Properties.Resources.ThreeDPieChart);
			cboStyle.Items.Add(Properties.Resources.ThreeDBarChart);
			cboStyle.Items.Add(Properties.Resources.FinanceChart);
			cboStyle.Items.Add(Properties.Resources.StackedAreaChart);

			// 「グラフの種類」コンボボックスの初期選択状態を設定します。
			cboStyle.SelectedIndex = 0;
		}

		private void cboStyle_SelectedIndexChanged(object sender, EventArgs e)
		{
			//「折れ線グラフ」選択時のみ、カスタムプロパティコンボボックスを有効にします。
			SetCustomProperties(cboStyle.SelectedIndex);
		}

		private void btnReport_Click(object sender, EventArgs e)
		{
			SectionReport rpt = new SectionReport();

			try
			{
				//「グラフの種類」コンボボックスにあわせて、プレビュー表示します。
				switch (cboStyle.SelectedIndex)
				{
					case 0: //  2D棒グラフ
						rpt.LoadLayout(XmlReader.Create(Properties.Resources.rpt2DBar));
						break;
					case 1: //  3D円グラフ
						rpt.LoadLayout(XmlReader.Create(Properties.Resources.rpt3DPie));
						//回転方向を設定します。
						if (cboCustom.SelectedIndex == 0)
						{
							((ChartControl)(rpt.Sections["Detail"].Controls["ChartSalesCategories"])).Series[0].Properties["Clockwise"] = true;
						}
						else
						{
							((ChartControl)(rpt.Sections["Detail"].Controls["ChartSalesCategories"])).Series[0].Properties["Clockwise"] = false;
						}
						break;
					case 2: // 3D棒グラフ
						rpt.AddAssembly(typeof(SQLiteConnection).Assembly);
						rpt.LoadLayout(XmlReader.Create (Properties.Resources.rpt3DBar));
						break;
					case 3: //  ファイナンスチャート
						rpt.LoadLayout(XmlReader.Create(Properties.Resources.rptCandle));
						break;
					case 4: //  積層エリアグラフ
						rpt.LoadLayout(XmlReader.Create (Properties.Resources.rptStackedArea));
						break;
				}

				arvMain.LoadDocument(rpt ?? new SectionReport());
			}
			catch (Exception ex)
			{
				MessageBox.Show(ex.ToString());
			}
		}

		private void SetCustomProperties(Int32 iGraphStype)
		{
			//選択候補をクリアします。
			cboCustom.Items.Clear();

			//選択候補を追加し、選択状態にします。
			switch (iGraphStype) 
			{
				case 1:	 // 2D円グラフ
					//可視状態にします。
					cboCustom.Visible = true;
					lblCustom.Visible = true;

					cboCustom.Items.Add(Properties.Resources.Clockwise);
					cboCustom.Items.Add(Properties.Resources.Counterclockwise);

					cboCustom.SelectedIndex = 1;

					//ラベルを設定します。
					lblCustom.Text = Properties.Resources.DirectionOfRotation;
					break;
					
				default:  //  その他
					//不可視状態にします。
					cboCustom.Visible = false;
					lblCustom.Visible = false;
					break;
			}

			Application.DoEvents();
		}
	}
}
